packages/components/src/base-control/index.tsx

```diff
@@ -104,10 +104,10 @@ export const BaseControl = ( {
  * 	</BaseControl>
  * );
  */
-export const VisualLabel: FunctionComponent< BaseControlVisualLabelProps > = ( {
+export const VisualLabel = ( {
 	className,
 	children,
-} ) => {
+}: WordPressComponentProps< BaseControlVisualLabelProps, 'span' > ) => {
```

Comment:
This marks it as polymorphic so we can use the `as` prop without TS complaining.

Reply:
Actually no this is [straightup wrong](https://github.com/WordPress/gutenberg/pull/42173#discussion_r932816116) 🤦 I'll fix it.

Reply:
Quickly tweaking this to pass along any additional props to the `StyledVisualLabel` fixed things up for me while testing.

<details>
<summary>FYI: Diff I tested with</summary>

```diff
diff --git a/packages/components/src/base-control/index.tsx b/packages/components/src/base-control/index.tsx
index d5b9dccf68..122577fdeb 100644
--- a/packages/components/src/base-control/index.tsx
+++ b/packages/components/src/base-control/index.tsx
@@ -107,9 +107,11 @@ export const BaseControl = ( {
 export const VisualLabel = ( {
 	className,
 	children,
+	...props
 }: WordPressComponentProps< BaseControlVisualLabelProps, 'span' > ) => {
 	return (
 		<StyledVisualLabel
+			{ ...props }
 			className={ classnames(
 				'components-base-control__label',
 				className

```
</details>

Reply:
Thanks, fixed in 03af77e.